manifests/config.pp

```diff
@@ -9,8 +9,13 @@
 class te_agent::config {
 
   $te_agent_file_ensure = $te_agent::agent ? {
-    true  => present,
-    false => absent,
+    true  => 'present',
```

Comment:
we are not supposed to quote states in resources,
however in variables is a gray area, im not sure that applies
https://www.puppet.com/docs/puppet/8/style_guide.html#quoting